src/esm_runscripts/workflow.py

```diff
@@ -1,49 +1,189 @@
 import sys, copy, os
 import esm_parser
 
+#import pdb
+
+class Workflow:
+    """A workflow class."""
+    default_phases = []
+    user_phases = []
+    always_run_with = []
+    first_task_in_queue = ""
+    last_task_in_queue = ""
+    next_run_triggered_by = ""
+
+    def __init__(self, phases, always_run_with=[]):
+        """
+        Create a new workflow.
+
+        Arguments:
+            phases -- List of workflow phases
+            always_run_with -- List of phases that precedes each phase
+        """
+        # TODO: NW call here the phase object ???
+        self.phases = phases
+        self.always_run_with = always_run_with
+
```

Comment:
```suggestion

    @property
```
Decorating this as a property means that you can use `self.num_phases_in_workflow` instead of `self.num_phases_in_workflow()`. I would also suggest that you change the name to `num_phases` since you are already in the workflow class/object.